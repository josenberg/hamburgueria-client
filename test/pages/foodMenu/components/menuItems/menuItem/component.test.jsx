import 'babel-polyfill';
import React from 'react';
import renderer from 'react-test-renderer';

import MenuItem from '../../../../../../app/pages/foodMenu/components/menuItems/menuItem';

test('MenuItem should render without issues', () => {
  const ingredients = [{
    displayName: 'Lanche',
    ingredients: [{
      displayName: 'Ingredient 1',
      price: 0.4,
    }, {
      displayName: 'Ingredient 2',
      price: 0.5,
    }],
  }];

  const item = {
    ingredients,
    displayName: 'X-Princesa',
  };

  const component = renderer.create(
    <MenuItem
      key="xprincesa"
      item={item}
      rules={[]}
      addToCart={() => {}}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
