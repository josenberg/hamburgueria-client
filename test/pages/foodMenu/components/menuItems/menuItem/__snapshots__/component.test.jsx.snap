// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without issues 1`] = `
<div
  className="grid"
  style={
    Object {
      "colGap": "0px",
      "display": "grid",
      "gridTemplateColumns": "1fr",
      "gridTemplateRows": "auto auto",
      "rowGap": "0px",
    }
  }
>
  <div
    className="grid-cell"
    style={
      Object {
        "display": "grid",
        "gridColumn": "auto",
        "gridRow": "auto",
      }
    }
  >
    <div
      className="grid"
      style={
        Object {
          "colGap": "0px",
          "display": "grid",
          "gridTemplateColumns": "100px 280px auto",
          "gridTemplateRows": "1",
          "margin": "5px auto",
          "rowGap": "0px",
          "width": "400px",
        }
      }
    >
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "1",
            "gridRow": "1",
          }
        }
      >
        <img
          alt="burger"
          src="test-file-stub"
          style={
            Object {
              "height": "80px",
            }
          }
        />
      </div>
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "2",
            "gridRow": "1",
          }
        }
      >
        <h2
          style={
            Object {
              "fontFamily": "'Lalezar', cursive",
              "margin": "5px 0 0 0",
            }
          }
        >
          X-Princesa
        </h2>
        <p
          style={
            Object {
              "margin": "-10px 0 0 0",
            }
          }
        >
          Lanche
        </p>
        <p
          style={
            Object {
              "margin": "0",
            }
          }
        >
          R$ NaN
        </p>
      </div>
      <div
        className="grid-cell"
        style={
          Object {
            "alignItems": "center",
            "display": "grid",
            "gridColumn": "3",
            "gridRow": "1",
          }
        }
      >
        <img
          alt="Add to cart"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</div>
`;
